bemvindo/tour.md

```diff
@@ -1,2 +1,306 @@
+
+Se o valor opcional for `nil`, a condição é `false` e o código entre chaves é pulado. Caso contrário, o valor opcional é desembrulhado e atribuído à constante após o `let`, tornando o valor desembrulhado disponível dentro do bloco de código.
+
+Outra maneira de lidar com valores opcionais é fornecer um valor padrão usando o operador `??`. Se o valor opcional estiver ausente, o valor padrão será utilizado.
+
+```swift
+let nickName: String? = nil
+let fullName: String = "John Appleseed"
+let informalGreeting = "Hi \(nickName ?? fullName)"
+```
+
+As instruções `switch` aceitam qualquer tipo de dado e uma variedade ampla de operações de comparação - elas não são limitadas a números inteiros e testes de igualdade.
+
+```swift
+let vegetable = "red pepper"
+switch vegetable {
+case "celery":
+    print("Adicione umas passas e faça um 'Ants on a Log'")
+case "cucumber", "watercress":
+    print("Isso daria um bom sanduíche")
+case let x where x.hasSuffix("pepper"):
+    print("É um \(x) apimentado?")
+default:
+    print("Tudo fica saboroso numa sopa.")
+}
+```
+
+>EXPERIMENTO
+>
+>Tente remover o caso `default`. Que erro você recebe?
+
+Repare como o `let` pode ser usado dentro de um padrão para atribuir à uma constante o valor que coincidiu com ele.
+
+Depois de executar o código dentro do `case` que foi utilizado, o programa sai do bloco de `switch`. A execução não continua para o próximo `case`, portanto não há necessidade de quebrar explicitamente o `switch` no final de cada bloco de `case`.
+
+Você usa `for-in` para iterar sobre itens de um dicionário, fornecendo um par de nomes a serem usados para cada par chave-valor. Dicionários são coleções não-ordenadas, portanto suas chaves e valores são iteradas numa ordem arbitrária.
+
+```swift
+let interestingNumbers = [
+    "Prime": [2, 3, 5, 7, 11, 13],
+    "Fibonacci": [1, 1, 2, 3, 5, 8],
+    "Square": [1, 4, 9, 16, 25]
+]
+var largest = 0
+for (kind, numbers) in interestingNumbers {
+    for number in numbers {
+        if number > largest {
+            largest = number
+        }
+    }
+}
+print(largest)
+```
+
+>EXPERIMENTO
+>
+>Adicione outra variável para armazenar que tipo de número foi o maiorm, bem como que número foi esse.
+
+
+Utilize `while` para repetir um bloco de código até que uma determinada condição seja alterada. A condição do laço de repetição pode estar no final também, garantindo que o laço será executado ao menos uma vez.
+
+```swift
+var n = 2
+while n < 100 {
+    n = n * 2
+}
+print(n)
+
+var m = 2
+repeat {
+    m = m * 2
+} while m < 100
+print(m)
+```
+
+Você pode manter um índice em um laço ou usando `..<` para fazer uma série de índices ou escrevendo uma inicialização explícita, condição e incremento. Estes dois laços fazem a mesma coisa:
+
+```swift
+var firstForLoop = 0
+for i in 0..<4 {
+    firstForLoop += i
+}
+print(firstForLoop)
+
+var secondForLoop = 0
+for var i = 0; i < 4; ++i {
```

Comment:
O livro original, por enquanto, se baseia numa ainda previa versão 2.2. Mas já foi aprovado que a versão final 2.2 irá gerar um _warning_ quando usar o "for C-like" ( `for(;;)` ). O que acha de já deixarmos isso como uma Nota do Tradutor? Algo como
```
> NOTA DO TRADUTOR
>
> Esta tradução se baseia na versão de dd/MM/yyyy, e após a disponibilização do livro original sob a CC by 4.0, foi aprovada a [SE-0007](https://github.com/apple/swift-evolution/blob/master/proposals/0007-remove-c-style-for-loops.md), que determina que esse estilo de laço de repetição será removido na versão 3.0 do Swift, e que na versão 2.2 final - o uso desse tipo de construção irá gerar um aviso de compilação - um _warning_. Por isso, o uso deste tipo de laço de repetição deve ser desencorajado."`

O que acha?